dask/bytes/core.py

```diff
@@ -194,56 +132,167 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
     10kB sample header and list of ``dask.Delayed`` objects or list of lists of
     delayed objects if ``fn`` is a globstring.
     """
-    if compression is not None and compression not in compress_files:
-        raise ValueError("Compression type %s not supported" % compression)
+    fs, names, myopen = get_fs_paths_myopen(urlpath, compression, 'rb',
+                                            None, **kwargs)
+    if len(names) == 0:
+        raise IOError("%s resolved to no files" % urlpath)
+    sizes = [fs.size(f) for f in names]
```

Comment:
Actually, even the standard lzmaffi/lzma library reads the whole file just the determine the size or to access any particular part of it - which is why the `get_xz_blocks` function should be useful, but it gives offsets in terms of raw bytes on disc.